src/compute/src/server.rs

```diff
@@ -636,13 +635,20 @@ impl<'w, A: Allocate + 'static> Worker<'w, A> {
                                         dataflow.as_of.as_ref().unwrap(),
                                     )
                                 });
-                            // We cannot reconcile subscriptions at the moment, because the response buffer is shared,
-                            // and to a first approximation must be completely reformed.
-                            let subscribe_free = dataflow
-                                .sink_exports
-                                .iter()
-                                .all(|(_id, sink)| !sink.connection.is_subscribe());
-                            if compatible && uncompacted && subscribe_free {
+
+                            // We cannot reconcile subscriptions at the moment, because the
+                            // response buffer is shared, and to a first approximation must be
+                            // completely reformed.
+                            let subscribe_free = dataflow.subscribe_ids().count() == 0;
```

Comment:
Nit: We don't have `is_empty` for iterators, should we use `peekable().next().is_none()` here? Probably doesn't matter since we have at most one export for each dataflow...

Reply:
I think just `next().is_none()` should work, no? I have no strong opinion on which is more readable, they both seem fine but not as nice as `is_empty`.

Reply:
Changed to `.next().is_none()` because it's theoretically more efficient, although as you say it doesn't matter as long as we don't have multi-export dataflows.

Reply:
Yes, still waking up from time off :)